components/match2/wikis/clashroyale/opponent_display_custom.lua

```diff
@@ -0,0 +1,140 @@
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+-- can not use Module:OpponentLibraries here due to circular requires
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display', {requireDevIfEnabled = true})
+
+local CustomOpponentDisplay = Table.copy(OpponentDisplay)
+
+CustomOpponentDisplay.BracketOpponentEntry = Class.new(
+	function(self, opponent, options)
+		self.content = mw.html.create('div'):addClass('brkts-opponent-entry-left')
+
+		if opponent.type == Opponent.team then
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo then
+			self:createPlayer(opponent.players[1])
+		elseif opponent.type == Opponent.duo then
+			self:createDuo(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+CustomOpponentDisplay.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
+CustomOpponentDisplay.BracketOpponentEntry.createLiteral = OpponentDisplay.BracketOpponentEntry.createLiteral
+CustomOpponentDisplay.BracketOpponentEntry.addScores = OpponentDisplay.BracketOpponentEntry.addScores
+
+function CustomOpponentDisplay.BracketOpponentEntry:createDuo(opponent)
+	local playerNode = CustomOpponentDisplay.PlayerBlockOpponent({
+		opponent = opponent,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
+
+function CustomOpponentDisplay.BracketOpponentEntry:createPlayer(player)
+	local playerNode = PlayerDisplay.BlockPlayer({
+		player = player,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
+
+function CustomOpponentDisplay.BlockOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.BlockOpponent, {maxDepth = 2})
+	local opponent = props.opponent
+
+	-- Default TBDs to not show links
+	props.showLink = Logic.nilOr(props.showLink, not Opponent.isTbd(opponent))
+
+	if opponent.type == Opponent.duo then
+		return CustomOpponentDisplay.PlayerBlockOpponent(props)
+	end
+
+	return OpponentDisplay.BlockOpponent(props)
+end
+
+function CustomOpponentDisplay.PlayerBlockOpponent(props)
+	local opponent = props.opponent
+
+	local playerNodes = Array.map(opponent.players, function(player)
+		return PlayerDisplay.BlockPlayer({
+			flip = props.flip,
+			overflow = props.overflow,
+			player = player,
+			showFlag = props.showFlag,
+			showLink = props.showLink,
+			showPlayerTeam = props.showPlayerTeam,
+			team = player.team,
+			abbreviateTbd = props.abbreviateTbd,
+		})
+	end)
+
+	if #opponent.players == Opponent.partySize(Opponent.solo) then
```

Comment:
Why this and not
`opponent.type == Opponent.solo`?

Reply:
because the function is also used in matchsummary submatches where you just have a list of players

Reply:
Looking at the context a bit closer... Is this branching even needed? Wouldn't the result in the `else` be effectively the same?

Reply:
I think so yeah